frontend/gateway/forwarder/forward.go

```diff
@@ -25,8 +26,8 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-func llbBridgeToGatewayClient(ctx context.Context, llbBridge frontend.FrontendLLBBridge, opts map[string]string, inputs map[string]*opspb.Definition, w worker.Infos, sid string, sm *session.Manager) (*bridgeClient, error) {
-	bc := &bridgeClient{
+func LLBBridgeToGatewayClient(ctx context.Context, llbBridge frontend.FrontendLLBBridge, exec executor.Executor, opts map[string]string, inputs map[string]*opspb.Definition, w worker.Infos, sid string, sm *session.Manager) (*BridgeClient, error) {
```

Comment:
[Why did you rename `func llbBridgeToGatewayClient`?](https://github.com/moby/buildkit/pull/4642#discussion_r1493012543)

Reply:
That is needed for breaking dependency loop

Reply:
What dependency loop? What _kind_ of dependency loop? How does renaming a function from unexported to exported help? I don't understand.